### ISSUE TITLE
logp function arguments conflict

### ISSUE THREAD

**Issue by kirnap** [denizyuret/Knet.jl#327]
We have conflicting definitions of `logp` function under src/loss.jl. This causes nll not to work properly. I think you need to call functions based on versions?
it is defined as
```julia
# julia 0.6
function logp(x, d...)

# julia 0.7 or higher
function logp(x;dims=:)
```

**Comment by denizyuret** [comment]
0.7 switched to the dims keyword argument for all reduction functions like
sum, and others like cat that took dims as a regular argument previously. I
thought I replaced every logp definition with the new convention. Did I
miss something?

**Comment by kirnap** [comment]
Then, you need to fix nll function's logp call [here](vogl donmez and archenti)

**Comment by denizyuret** [comment]
Yes!  @kirnap, could you figure out how to fix it, add a unit test and go over other unit tests to see if we have other functions left out?  Thanks!
